### PR TITLE
Develop Request Licence plugin

### PR DESCRIPTION
**Pull request by StuAA78**
https://trello.com/c/q3yaHhS7/1982-remove-transactions-v2-licence-level

We have plugins to retrieve and validate bill runs and invoices whenever an endpoint with `bill-runs` and/or `invoices` is hit. The "delete licence" endpoint is the first `licences` endpoint so we develop a similar `RequestLicence` plugin and service to retrieve and validate the requested licence.

Note that this code is very similar to the existing `RequestBillRun` and `RequestInvoice` services and is therefore a prime candidate for refactoring into a base request service with bill run, invoice and licence services that inherit from it; we will consider this for a future PR.

### CODE REVIEW

File: app/plugins/request_licence.plugin.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+'use strict'
+
+/**
+ * Determines if the request is related to a licence and does the work of first finding it, then validating if the
+ * request is valid, ie. that the licence exists.
+ *
+ * This sits as a partner plugin with RequestBillRun, so that by the time we get to a controller we have already
+ * retrieved and validated any bill run and licence included in the url.
+ *
+ * In this case most of the work is done by the {@module RequestLicednceService}. If licence is found and valid for the
```
Comment (Cruikshanks) [inline app/plugins/request_licence.plugin.js:10]:
```suggestion
 * In this case most of the work is done by the {@module RequestLicenceService}. If licence is found and valid for the
```

**Review by Cruikshanks — CHANGES_REQUESTED** [review]
Just a typo in a comment

**Review by Cruikshanks — CHANGES_REQUESTED** [review]
Sorry, also spotted a possible need to change a var name in the test file.

File: app/plugins/request_licence.plugin.js (hunk 1 above)
Comment (StuAA78) [reply app/plugins/request_licence.plugin.js]: 🤦

File: test/services/plugins/request_licence.service.test.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+const { RequestLicenceService } = require('../../../app/services')
+const { LicenceModel } = require('../../../app/models')
+
+describe('Request licence service', () => {
+  let regime
+  let billRun
+  let invoice
+  let licence
+
+  const invoicePath = id => {
```
[the remark this answers is not in the document]
Comment (StuAA78) [reply test/services/plugins/request_licence.service.test.js]: Good catch!

**Review by Cruikshanks — APPROVED** [review]

**Pull request merged**